tgui/packages/tgui/interfaces/MfdPanels/FiremissionPanel.tsx

```diff
@@ -0,0 +1,580 @@
+  { min: -6, max: 0, values: ['-6', '-5', '-4', '-3', '-2', '-1', '0', '-'] },
+  { min: 0, max: 6, values: ['-', '0', '1', '2', '3', '4', '5', '6'] },
+  { min: 0, max: 6, values: ['-', '0', '1', '2', '3', '4', '5', '6'] },
+];
+
+const OffsetOverview = (
+  props: MfdProps & { fm: CasFiremission; equipment: DropshipEquipment }
+) => {
+  const weaponFm = props.fm.records.find(
+    (x) => x.weapon === props.equipment.mount_point
+  );
+  if (weaponFm === undefined) {
+    return <>error</>;
+  }
+  const ammoConsumption = weaponFm.offsets
+    .map((x) => (x !== '-' ? props.equipment.burst ?? 0 : 0))
+    .reduce((accumulator, currentValue) => accumulator + currentValue, 0);
+  return (
+    <>
+      <Stack.Item className="FireMissionOffsetLabel">
+        {props.equipment.shorthand} {props.equipment.mount_point}
+      </Stack.Item>
+      <Stack.Item className="FireMissionOffsetLabel">
+        {props.equipment.ammo} / {props.equipment.max_ammo}
+      </Stack.Item>
+      <Stack.Item className="FireMissionOffsetLabel">
+        {ammoConsumption}
+      </Stack.Item>
+    </>
+  );
+};
+
+const OffsetDetailed = (
+  props: MfdProps & {
+    fm: CasFiremission;
+    equipment: DropshipEquipment;
+  }
+) => {
+  const availableGimbals = gimbals[props.equipment.mount_point];
+  const weaponFm = props.fm.records.find(
+    (x) => x.weapon === props.equipment.mount_point
+  );
+  if (weaponFm === undefined) {
+    return <>error</>;
+  }
+  const ammoConsumption = weaponFm.offsets
+    .map((x) => (x === '-' ? props.equipment.burst : 0))
+    .reduce(
+      (accumulator, currentValue) => (accumulator ?? 0) + (currentValue ?? 0),
+      0
+    );
+  return (
+    <>
+      <Stack.Item className="FireMissionOffsetLabel">
+        {props.equipment.shorthand} {props.equipment.mount_point}
+      </Stack.Item>
+      <Stack.Item className="FireMissionOffsetLabel">
+        {props.equipment.ammo} / {props.equipment.max_ammo} using{' '}
+        {ammoConsumption} per run.
+      </Stack.Item>
+      <Stack.Item className="FireMissionOffsetLabel">
+        {availableGimbals.min} to {availableGimbals.max}
+      </Stack.Item>
+      <Stack.Item className="FireMissionOffsetLabel">
+        {(props.equipment.firemission_delay ?? 0) - 1}
+      </Stack.Item>
+    </>
+  );
+};
+
+const FMOffsetError = (
+  props: MfdProps & {
+    fm: CasFiremission;
+    equipment: DropshipEquipment;
+    displayDetail?: boolean;
+  }
+) => {
+  return (
+    <Stack vertical className="FireMissionStack">
+      {props.displayDetail ? (
+        <OffsetDetailed
+          fm={props.fm}
+          panelStateId={props.panelStateId}
+          equipment={props.equipment}
+        />
+      ) : (
+        <OffsetOverview
+          fm={props.fm}
+          panelStateId={props.panelStateId}
+          equipment={props.equipment}
+        />
+      )}
+      <Stack.Item height="25px" />
+      <Divider />
+      <Stack.Item>
+        Unable to set firemission offsets.
```

Comment:
This screen doesn't really fit if both weapons have this error:
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/a52cbc3c-f1d5-4089-acb5-b9222ed4eebf)

Reply:
![image](https://github.com/cmss13-devs/cmss13/assets/101871009/19b0d795-d5ff-4840-96ad-04684d33cb75)
I've fixed the width, it seems good